### PR TITLE
[Components with more than 1 service] LPS-179177 Refactor ScreenNavigationCategory components from Commerce modules part 1

### PR DESCRIPTION
**Pull request by daiazius**

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:startup

**Comment by daiazius** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
There were mass failures I can reproduce. Will rerun tests.

**Comment by Kvale1733** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:commerce

**Comment by Kvale1733** [comment]
@daiazius No regressions found.

**Review by rafaprax — APPROVED** [review]

**Comment by daiazius** [comment]
Hey, I just changed the commit messages, no changes in the code.

**Comment by rafaprax** [comment]
LGTM

**Review by rafaprax — APPROVED** [review]

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:commerce

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by tinatian** [comment]
@dantewang Please take this over too.

Thanks
TIna

File: modules/apps/commerce/commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/frontend/taglib/servlet/taglib/AccountEntryCommerceChannelDefaultsScreenNavigationEntry.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	property = "screen.navigation.entry.order:Integer=10",
+	service = ScreenNavigationEntry.class
+)
+public class AccountEntryCommerceChannelDefaultsScreenNavigationEntry
+	extends AccountEntryCommerceChannelDefaultsScreenNavigationCategory
+	implements ScreenNavigationEntry<AccountEntry> {
+
+	@Override
+	public String getEntryKey() {
+		return AccountEntryScreenNavigationEntryConstants.
```
Comment (dantewang) [inline modules/apps/commerce/commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/frontend/taglib/servlet/taglib/AccountEntryCommerceChannelDefaultsScreenNavigationEntry.java:65]: @daiazius Please make `getEntryKey()` delegate to `getCategoryKey()` when they return the same constant.

**Review by dantewang — CHANGES_REQUESTED** [review]
@daiazius Please see my comment, thanks.

File: modules/apps/commerce/commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/frontend/taglib/servlet/taglib/AccountEntryCommerceChannelDefaultsScreenNavigationEntry.java (hunk 1 above)
Comment (tinatian) [reply modules/apps/commerce/commerce-account-web/src/main/java/com/liferay/commerce/account/web/internal/frontend/taglib/servlet/taglib/AccountEntryCommerceChannelDefaultsScreenNavigationEntry.java:65]: @dantewang I asked them to use constants, as using delegate will introduce another reference, and it does not make sense to have that reference.

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:commerce

**Comment by daiazius** [comment]
ci:test:startup

**Review by dantewang — APPROVED** [review]

**Comment by zhangyan0701** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@daiazius regression found:

Steps to reproduce

1. Create a minium site called "minium"
2. Go to applications menu > commerce > channels
3. click in the site minium

Actual result:
Only the tabs are present but any other information is displayed and an error appears in the log:
```
2023-04-05 16:34:27.059 ERROR [http-nio-8080-exec-2][IncludeTag:128] Current URL /group/test-minium/~/control_panel/manage?p_p_id=com_liferay_commerce_channel_web_internal_portlet_CommerceChannelsPortlet&p_p_lifecycle=0&p_p_state=maximized&p_p_auth=Rz5COYj4&_com_liferay_commerce_channel_web_internal_portlet_CommerceChannelsPortlet_backURL=http%3A%2F%2Flocalhost%3A8080%2Fgroup%2Ftest-minium%2F%7E%2Fcontrol_panel%2Fmanage%3Fp_p_id%3Dcom_liferay_commerce_channel_web_internal_portlet_CommerceChannelsPortlet%26p_p_lifecycle%3D0%26p_p_state%3Dmaximized%26p_p_auth%3DRz5COYj4&_com_liferay_commerce_channel_web_internal_portlet_CommerceChannelsPortlet_commerceChannelId=44177&_com_liferay_commerce_channel_web_internal_portlet_CommerceChannelsPortlet_mvcRenderCommandName=%2Fcommerce_channels%2Fedit_commerce_channel generates exception: java.lang.NullPointerException
java.lang.NullPointerException: null
	at com.liferay.commerce.channel.web.internal.frontend.taglib.servlet.taglib.CommerceChannelGeneralScreenNavigationCategory.getLabel(CommerceChannelGeneralScreenNavigationCategory.java:49) ~[bundleFile:?]
	at org.apache.jsp.screen_005fnavigation.page_jsp._jspService(page_jsp:405) ~[?:?]
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:111) ~[jasper.jar:9.0.73]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:596) ~[servlet-api.jar:4.0.FR]
```

Expected result:
Informations of the site channels should display without errors.

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
@daiazius Please see https://github.com/liferay-core-infra/liferay-portal/pull/4141, fix the issue and resend.

**Pull request closed without merging**